### PR TITLE
Upgrade to Python 3.9

### PR DESCRIPTION
**Pull request by cyyever**
### Description

Update project files to use Python 3.9 as the minimum version and apply ruff fix

### CODE REVIEW

**Comment by andife** [comment]
Many thanks for the adjustments.

I wonder if https://pypi.org/project/refurb/ could also be of interest, or if this is already integrated into ruff?

**Comment by cyyever** [comment]
@andife I don't know, but it can be added into lintrunner

**Comment by justinchuby** [comment]
> Many thanks for the adjustments.
> 
> I wonder if https://pypi.org/project/refurb/ could also be of interest, or if this is already integrated into ruff?

Ruff has implemented some refurb rules iirc

**Review by justinchuby — APPROVED** [review]

**Comment by andife** [comment]
@cyyever "applying lintrunner -a" solves some pull requests comments e.g. a) sorting of imports and b) List[List[int]], =>          list[list[int]],

File: pyproject.toml [hunk 1]
```diff
@@ -97,7 +97,7 @@ ignore_errors = true
 
 [tool.black]
 # NOTE: Do not create an exclude list. Edit .lintrunner.toml instead
-target-version = ["py38", "py39", "py310", "py311"]
+target-version = ["py39", "py310", "py311"]
```
Comment (andife) [inline pyproject.toml:100]: should we add py312? py313?

**Pull request merged**